server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -641,6 +646,56 @@ private String getPrettyName() throws IOException {
         }
     }
 
+    /**
+     * Returns the lines from /proc/meminfo as a workaround for JDK bugs that prevent retrieval of total system memory
+     * on some Linux variants such as Debian8.
+     */
+    @SuppressForbidden(reason = "access /proc/meminfo")
+    List<String> readProcMeminfo() throws IOException {
```

Comment:
I'm not thrilled by everything that happens here, but by default, the value is cached so that this method shouldn't be called more than once per second and probably less frequently on systems with appropriate monitoring intervals.